File: Templates/mod/sysadmin/typescript/src/sysadmin.ts
type GregRuntimeHooks = {
  mod_on_init?: () => void;
  mod_on_update?: (deltaTime: number) => void;
  mod_on_shutdown?: () => void;
};

let elapsed = 0;
const heartbeatSeconds = 10;

export const hooks: GregRuntimeHooks = {
  mod_on_init: () => {
    console.log("[gregCore][Sysadmin][TS] initialized");
  },
  mod_on_update: (deltaTime) => {
    elapsed += deltaTime;
    if (elapsed >= heartbeatSeconds) {
      elapsed = 0;
      console.log("[gregCore][Sysadmin][TS] heartbeat");
    }
  },
  mod_on_shutdown: () => {
    console.log("[gregCore][Sysadmin][TS] shutdown");
  }
};

File: gregTemplates/mod/sysadmin/lua/sysadmin.lua
local mod = {
  id = "lua:sysadmin-example",
  name = "Sysadmin Lua Example",
  version = "0.1.0"
}

local heartbeatSeconds = 10.0
local elapsed = 0.0

function mod_on_init()
  print("[gregCore][Sysadmin][Lua] initialized")
end

function mod_on_update(delta)
  elapsed = elapsed + delta
  if elapsed >= heartbeatSeconds then
    elapsed = 0.0
    print("[gregCore][Sysadmin][Lua] heartbeat")
  end
end

function mod_on_shutdown()
  print("[gregCore][Sysadmin][Lua] shutdown")
end

return mod
